Firestore/core/src/bundle/bundle_serializer.h

```diff
@@ -19,25 +19,81 @@
 
 #include <string>
 #include <utility>
+#include <vector>
 
+#include "Firestore/core/src/bundle/bundle_document.h"
 #include "Firestore/core/src/bundle/bundle_metadata.h"
+#include "Firestore/core/src/bundle/bundled_document_metadata.h"
+#include "Firestore/core/src/bundle/named_query.h"
+#include "Firestore/core/src/core/core_fwd.h"
+#include "Firestore/core/src/model/resource_path.h"
 #include "Firestore/core/src/model/snapshot_version.h"
+#include "Firestore/core/src/remote/serializer.h"
 #include "Firestore/core/src/util/read_context.h"
 #include "nlohmann/json.hpp"
 
 namespace firebase {
 namespace firestore {
 namespace bundle {
 
+class JsonReader : public util::ReadContext {
```

Comment:
Consider adding a class comment to this type that indicates generally how these functions work. Ideally these would all have comments, but they're all mostly the same except for types, so that's probably overkill. A single class comment that describes how `Require` methods will `Fail` if the key is not present or has the wrong type and then return a default value would be enough.